pkg/config/migration/multi_account.go

```diff
@@ -0,0 +1,124 @@
+package migration
+
+import (
+	"fmt"
+
+	"github.com/cli/go-gh/v2/pkg/config"
+)
+
+type CowardlyRefusalError struct {
+	Reason string
+}
+
+func (e CowardlyRefusalError) Error() string {
+	// Consider whether we should add a call to action here like "open an issue with the contents of your redacted hosts.yml"
+	return fmt.Sprintf("cowardly refusing to continue with multi account migration: %s", e.Reason)
+}
+
+var hostsKey = []string{"hosts"}
+
+// This migration exists to take a hosts section of the following structure:
+//
+//	github.com:
+//	  user: williammartin
+//	  git_protocol: https
+//	  editor: vim
+//	github.localhost:
+//	  user: monalisa
+//	  git_protocol: https
+//
+// We want this to migrate to something like:
+//
+// github.com:
+//   active_user: williammartin
+//	 users:
+//	   williammartin:
+//	     git_protocol: https
+//	     editor: vim
+//
+// github.localhost:
+//	 active_user: monalisa
+//     users:
+//	     monalisa:
+//		   git_protocol: https
+//
+// The reason for this is that we can then add new users under a host.
+
+type MultiAccount struct{}
+
+func (m MultiAccount) Do(c *config.Config) (bool, error) {
+	hostnames, err := c.Keys(hostsKey)
+	// [github.com, github.localhost]
+	if err != nil {
+		return false, CowardlyRefusalError{"couldn't get hosts configuration"}
+	}
+
+	// If there are no hosts then it doesn't matter whether we migrate or not,
+	// so lets avoid any confusion and say there's no migration required.
+	if len(hostnames) == 0 {
+		return false, nil
+	}
+
+	// If there is an active user set for the first host then we must have already
```

Comment:
Want to draw attention to this decision.

Reply:
Makes sense to me. I am curious if we add in a version what places in `gh` would actually need to change? I can only think of maybe 1 or 2 off the top of my head.

Reply:
I would think putting it in the top level of `config.yml` is the right spot. Any change to the structure of `config.yml` or `hosts.yml` would trigger us bumping that version. We should treat them as a single config structure despite being in two separate files.

Reply:
Ooh, that's super obvious. I like it, and it makes me happy that I decided the `Migration` should work with the entire config rather than hardcoding it for just hosts.